src/analysis/inner_product_space/adjoint.lean

```diff
@@ -254,6 +274,50 @@ begin
   exact ext_inner_right 𝕜 (λ y, by simp only [adjoint_inner_left, h x y])
 end
 
+lemma is_self_adjoint_iff_eq_adjoint (A : E →L[𝕜] E) :
+  is_self_adjoint (A : E →ₗ[𝕜] E) ↔ A = A.adjoint :=
+by simp_rw [is_self_adjoint, coe_coe, ← eq_adjoint_iff]
+
+lemma _root_.inner_product_space.is_self_adjoint.eq_adjoint {A : E →L[𝕜] E}
+  (hA : is_self_adjoint (A : E →ₗ[𝕜] E)) : A = A† :=
+by rwa is_self_adjoint_iff_eq_adjoint at hA
+
+lemma _root_.inner_product_space.is_self_adjoint.conj_adjoint {T : E →L[𝕜] E}
```

Comment:
Since `S††` is not defeq to `S` maybe we should add a variant of this lemma stating `is_self_adjoint (S† ∘L T ∘L S : F →ₗ[𝕜] F)`?